src/main/scala/edu/twitter/evaluation/ModelEvaluator.scala

```diff
@@ -0,0 +1,116 @@
+package edu.twitter.evaluation
```

Comment:
I believe this class constructor has to take the same training and testing data used to train the model!!!?

Reply:
I agree. As I mentioned I will add the method for that but we need also the ability to evaluate a given model with labeled data in order to use something like `textblob` in the future for evaluation.

Reply:
we can make a method that takes data as a parameter

Reply:
We can consider something like that. but I guess from encapsulation perspective we can leave the implemented `evaluate` method.

Reply:
We need to distinguish between two different times where the evaluation is required:
1) Evaluating the model after it's build using the test data.
2) Evaluating a given model using a specific labeled data set from ( textblob or google cloud api) in order to compare it with other models.

The first part is more related to the building process and in my opinion, it should be moved to the ModelBuilder. Fortunately, as we are mainly concerned with the accuracy for this task, each of the two models that we are building has its already implemented method for the evaluation:
- evaluateEachIteration() for the Gradient Boosted Trees Model
- evaluate() for the Neural Network Model

This way, this class would be responsible for the second task only.

Reply:
Totally agree, this idea will be much cleaner in the implementation :+1:

Reply:
I don't know why but evaluateEachIteration() doesn't work as I was expecting, I will check it again later. 

For now, could you please consider using MulticlassMetrics? I have tried it and it looks good.  
https://spark.apache.org/docs/2.2.0/mllib-evaluation-metrics.html#tab_scala_1